firmware/Core/Src/telecommands/telecommand_parser.c

```diff
@@ -279,7 +279,31 @@ uint8_t TCMD_parse_full_telecommand(const char tcmd_str[], TCMD_TelecommandChann
         return 90;
     }
 
-    // Reached the end of the telecommand parsing. Thus, success. Fill the output struct.
+    // ensure the corrent number of args are provided
+    int32_t num_args_expected = TCMD_telecommand_definitions[tcmd_idx].number_of_args;
```

Comment:
`int16_t` would be more appropriate (or even `int8_t`). Please use `const` keyword to mark variable as immutable.